nexus/types/src/deployment/planning_input.rs

```diff
@@ -333,6 +333,32 @@ pub struct SledDetails {
     pub resources: SledResources,
 }
 
+impl SledDetails {
+    /// Returns `Some(reason)` if the sled needs its zones to be expunged,
+    /// based on the policy and state.
+    pub fn needs_zone_expungement(&self) -> Option<ZoneExpungeReason> {
```

Comment:
This is pretty nitty, but: this behavior seems really specific to planning and it's a little surprising to me that it would live in nexus/types and under PlanningInput.  Why wouldn't the planner check this condition explicitly?

Reply:
I see your point here. I think originally, I was thinking of sharing this across planning and some other code -- but that's not necessary for now. I'll move it to the planner.

Reply:
Done.